CRM/Upgrade/Snapshot.php

```diff
@@ -57,15 +60,16 @@ public static function getActivationIssues(): array {
       static::$activationIssues = [];
       foreach ($limits as $table => $limit) {
         try {
-          $count = CRM_Core_DAO::singleValueQuery("SELECT count(*) FROM `{$table}`");
+          // Use select MAX(id) rather than COUNT as COUNT is slow on large databases.
+          $max = CRM_Core_DAO::singleValueQuery("SELECT MAX(id) FROM `{$table}`");
         }
-        catch (\Exception $e) {
-          $count = 0;
+        catch (CRM_Core_Exception $e) {
```

Comment:
These are usually PEAR_Exceptions

Reply:
@demeritcowboy thrown from the `CRM_Core_DAO::executeQuery` ?

Reply:
Yes.

You can simulate it with something like `cv ev 'try { CRM_Core_DAO::singleValueQuery("SELECT begargle FROM civicrm_contact LIMIT 1"); } catch (\Exception $e) { echo get_class($e); }'`

Reply:
ok - so on one hand I should possibly fix to catch the PEAR_EXCEPTION - but I'm also  inclined to add a new exception `CRM_DatabaseQuery_Exception` that extends `CRM_Core_Exception` - that way we can a) catch all civi exceptions within civi & also know that it is a db exception